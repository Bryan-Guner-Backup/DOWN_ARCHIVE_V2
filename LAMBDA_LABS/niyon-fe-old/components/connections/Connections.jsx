import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import TopSection from '../TopSection';
import ProfileList from './ProfileList';
import withUserData from '../containers/withUserData';
import ConnectionButtons from './ConnectionsButtons';
import ConnectionRequests from './ConnectionRequests';

const Wrapper = styled.div`
    width: 100%;
    display: flex;
    flex-direction: column;
`;

function Connections(props) {
  const connectionsLength = props.connectionsAll ? props.connectionsAll.length : 0;

  return (
    <Wrapper>
      <TopSection
        buttons={(
          <ConnectionButtons
            numOfConnections={connectionsLength}
          />
)}
        src="/static/friends-online.svg"
      />
      <ConnectionRequests requests={props.connectionsReceived.received || []} connections={props.connectionsAll || []} />
      <ProfileList title="My Connections" users={props.connectionsAll || []} />
    </Wrapper>
  );
}

Connections.propTypes = {
  connectionsAll: PropTypes.arrayOf(PropTypes.shape()).isRequired,
  connectionsReceived: PropTypes.arrayOf(PropTypes.shape()).isRequired,
};

export default withUserData(Connections);
