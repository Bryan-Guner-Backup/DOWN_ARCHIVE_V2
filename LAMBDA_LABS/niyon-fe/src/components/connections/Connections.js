import React from 'react'
import Paper from '@material-ui/core/Paper'
import SwipeTabsConnections from './SwipeTabsConnections'

export default function Connections (props) {
  // console.log('connection props', props)
  return (
    <div className="connections" data-test="connections">
      <div data-test="second-wrapper">
        <Paper className="paper" elevation={1} data-test="paper">
          <h1 className="container-header" data-test="my-connections">My Connections ({props.sumConnections})</h1>
          <div>
            <SwipeTabsConnections />
          </div>
        </Paper>
      </div>
    </div>
  )
};
