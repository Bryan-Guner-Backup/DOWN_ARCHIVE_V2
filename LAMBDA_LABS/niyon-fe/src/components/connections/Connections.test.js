import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Connections from './Connections'
import findByTestAttr from '../../tests/utils'

const setUp = (props = {}) => {
  const wrapper = shallow(<Connections { ...props } />)
  return wrapper
}

describe('<Connections /> component testing', () => {
    const testProps = {
      sumConnections: 1
    }
    let component;
    beforeEach(() => {
      component = setUp({...testProps});
    });
    it('should match snapshot tests', () => {
        expect(toJSON(component)).toMatchSnapshot();
    })

    it('should render <Connections /> component correctly', () => {
        const connectionsWrapper = findByTestAttr(component, 'connections');
        expect(connectionsWrapper.length).toBe(1);
    });

    it('should render a second container <Div /> correctly', () => {
         const secondWrapper = findByTestAttr(component, 'second-wrapper');
         expect(secondWrapper.exists()).toBe(true);
    });
    it('should render <Paper /> component correctly', () => {
      const wrapper = findByTestAttr(component, 'paper');       
        expect(wrapper.exists()).toBe(true);
    });    
    
    it('should render mani title in the component', () => {
            const sum = testProps.sumConnections;
            const h1 = findByTestAttr(component, 'my-connections');
            expect(h1.length).toBe(1);
            expect(h1.text().trim()).toEqual(`My Connections (${sum})`);
    }); 
    
    it('should render <SwipeTabsConnections /> component correctly', () => {
         expect(component.find('SwipeTabsConnections').exists()).toBe(true);
    });        
});
