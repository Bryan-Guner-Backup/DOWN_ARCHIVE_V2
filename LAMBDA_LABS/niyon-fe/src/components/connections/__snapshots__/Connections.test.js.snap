// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Connections /> component testing should match snapshot tests 1`] = `
<div
  className="connections"
  data-test="connections"
>
  <div
    data-test="second-wrapper"
  >
    <WithStyles(ForwardRef(Paper))
      className="paper"
      data-test="paper"
      elevation={1}
    >
      <h1
        className="container-header"
        data-test="my-connections"
      >
        My Connections (
        1
        )
      </h1>
      <div>
        <SwipeTabsConnections />
      </div>
    </WithStyles(ForwardRef(Paper))>
  </div>
</div>
`;
